internal/host/host.go

```diff
@@ -419,3 +414,66 @@ func (m *Manager) reportInstallationMetrics(ctx context.Context, h *models.Host,
 		m.metricApi.ReportHostInstallationMetrics(log, cluster.OpenshiftVersion, h, previousProgress, CurrentStage)
 	}
 }
+
+func (m *Manager) AutoAssignRole(ctx context.Context, h *models.Host, db *gorm.DB) error {
+	// select role if needed
+	if h.Role == models.HostRoleAutoAssign {
+		return m.autoRoleSelection(ctx, h)
+	}
+	return nil
+}
+
+func (m *Manager) selectRole(ctx context.Context, h *models.Host) (models.HostRole, error) {
```

Comment:
Small nit: since the calls are AutoAssignRole -> autoRoleSelection -> selectRole, can you make the order they appear in the code the same?  Just makes reading the code a bit easier (i.e., move autoRoleSelection here right after AutoAssignRole).